modules/Tvpapi/resources/tvpapiAnalytics.js

```diff
@@ -71,6 +84,30 @@
 			});
 		},
 
+		handleLoad:function(_this) { // chanaka 
+			if (_this.licensedLinksRecieved && _this.loadRecieved){
+				_this.sendMediaMark('load');
+				_this.loadRecieved = false;
+			}
+		},
+
+		handleFirstPlay:function(_this) { // chanaka 
+			if (_this.licensedLinksRecieved && _this.firstplayRecieved){
+				_this.didFirstPlay = true;
+				_this.sendMediaMark('first_play');
+				_this.firstplayRecieved = false;
+			}
+		},
+
+		handlePlay:function(_this) { // chanaka 
+			if (_this.licensedLinksRecieved && _this.playRecieved){
+				_this.isPlaying = true;
+				_this.startMediaHitInterval();
+				_this.sendMediaMark('play');
+				_this.playRecieved = false;
+			}
+		},
+
```

Comment:
if code uses instance members why put it outside and pass this to it, move this to inside the plugin - but before doing anything please refer to previous comment as this might be redundent